### PR TITLE
add info about git and git-lfs to docs

### PR DESCRIPTION
**Pull request by rokroskar**
Improves the user documentation including:

* info about how to install into an isolated environment with conda
* some first instructions to use git-lfs
* git tips
* add a note about `renku rerun` to the first steps 

addresses #452 
addresses #343

### CODE REVIEW

File: docs/user/cli-installation.rst [hunk 1]
```diff
@@ -51,6 +51,69 @@ If you want to have Renku as a project dependency, you may install it using
   pipenv shell
 
 
+<<<<<<< HEAD
```
Comment (leafty) [inline docs/user/cli-installation.rst]: Unresolved merge conflict here

Comment (rokroskar) [reply docs/user/cli-installation.rst]: indeed, thanks

Comment (rokroskar) [reply docs/user/cli-installation.rst]: indeed, thanks!

**Comment by leafty** [comment]
```
user/cli-installation.rst:57:conda:["coda", "condo", "conga", "Ronda", "Honda", "Fonda", "Vonda", "Zonda", "anaconda", "Golconda", "Gioconda", "Corenda", "contd", "condemn"]
user/cli-installation.rst:63:conda:["coda", "condo", "conga", "Ronda", "Honda", "Fonda", "Vonda", "Zonda", "anaconda", "Golconda", "Gioconda", "Corenda", "contd", "condemn"]
user/cli-installation.rst:101:conda:["coda", "condo", "conga", "Ronda", "Honda", "Fonda", "Vonda", "Zonda", "anaconda", "Golconda", "Gioconda", "Corenda", "contd", "condemn"]
user/firststeps.rst:407:hyperparameters:["hyper parameters", "hyper-parameters", "parameterless", "parameters", "parameterizes", "parameterizations", "parameterization", "hyperthyroidism"]
user/git_tips.rst:48:zhvelo:["Lovell", "Zhivago", "Havel"]
```

**Comment by rokroskar** [comment]
you mean I should do the spellcheck *before* pushing??

**Comment by rokroskar** [comment]
I realized it was too complicated to explain the submodules etc so I'm trying to quickly code up something in `renku-python`. If that doesn't turn out to be a good solution, we will need to document everything about how renku uses submodules (not a bad thing to do in any case)

**Comment by rokroskar** [comment]
@leafty did you see the isolated conda environment install instructions? anything to add there?

**Comment by leafty** [comment]
It's the working set of instructions, no? I remember trying this in a anaconda3 container.

File: docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -54,4 +56,5 @@ Versioning
 vertices
 workflow
 workflows
+zhvelo
```
Comment (leafty) [inline docs/spelling_wordlist.txt]: Nooooo :scream:

File: docs/user/git_tips.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    commit f346c1f36f211840b22a63e7fc2c9bfb52616212
+    Author: Rok Roskar <roskarr@ethz.ch>
+    Date:   Wed Oct 3 22:26:38 2018 +0200
+
+        renku dataset create zhvelo
+
+
+Now lets say you decide your data imports and everything that followed was
+wrong and you want to go back to "renku dataset create zhvelo", i.e.
```
Comment (leafty) [inline docs/user/git_tips.rst]: Use code here to avoid spellcheck

File: docs/spelling_wordlist.txt (hunk 2 above)
Comment (rokroskar) [reply docs/spelling_wordlist.txt]: I know, it felt a little bit dirty... :)

**Comment by rokroskar** [comment]
@leafty yes I tested it on my mac, it works well.

**Pull request merged**